src/main/java/com/google/cloud/tools/Action.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools;
+
+import com.google.cloud.tools.Option.Type;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+/**
+ * Contains common members and methods to all {@link Action} implementations.
+ */
+// TODO(joaomartins): Come up with a better name for this. Action too overloaded.
+public abstract class Action {
+
+  /**
+   * Flags sent by the client.
+   */
+  protected Map<Option, String> flags;
```

Comment:
May be subsumed by Patrick's issues with flags, but in general mutable non-private fields can change at unexpected times. For example a subclass could change a value in this field or replace it with a completely different map.